### PR TITLE
MODINVSTOR-663: Log failed domain events to internal table.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
## Approach:

* Create new `notification_sending_error` table with following structure:
  * id;
  * topicName;
  * partitionKey;
  * payload;
  * error;
  * incidentDateTime;
* Introduce `FailureHandler` interface that injected into domain events publisher class
* Add LogToDbTable default implementation for `FailureHandler`.

### CODE REVIEW

File: src/main/java/org/folio/services/domainevent/LogToDbFailureHandler.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import java.util.Date;
+import java.util.Map;
+import java.util.UUID;
+import org.folio.persist.NotificationSendingErrorRepository;
+import org.folio.persist.entity.NotificationSendingError;
+
+public class LogToDbFailureHandler implements FailureHandler {
+  private final NotificationSendingErrorRepository repository;
+
+  public LogToDbFailureHandler(NotificationSendingErrorRepository repository) {
```
Comment (hjiebsco) [inline src/main/java/org/folio/services/domainevent/LogToDbFailureHandler.java:17]: Looks like `public` modifier can be omitted to reduce visibility.

**Review by hjiebsco — APPROVED** [review]
Assuming sys admin/operator needs to monitor the new table.

**Review by pfilippovEPAM — APPROVED** [review]
Looks good to me

File: src/main/java/org/folio/services/domainevent/LogToDbFailureHandler.java (hunk 1 above)
Comment (bohdan-suprun) [reply src/main/java/org/folio/services/domainevent/LogToDbFailureHandler.java]: Changed, thanks

**Pull request merged**